test/main.cpp

```diff
@@ -59,6 +59,118 @@ TEST_CASE("Manipulate cursor position", "[cursor]") {
     }
 }
 
+/*
+ * Testing the cursor position manipulation using the set_cursor_position frame
+ * The cursor used is the open_cursor 
+ * The test consists of 3 parts testing both X,Y out of bounds and each seperatly.
+ *
+ * The test will try to move the cursor to (130,50),(50,180),(130,180)
+ * since either X,Y or both are out of bounds the cursor,
+ * will remain on the start position for the out of bounds position
+ */
+TEST_CASE("Manipulate cursor position out of bounds", 
+          "[cursor, position, internal_communication]") {
+
+    // The bus itself doesn't take any constructor arguments
+    r2d2::mock_comm_c mock_bus;
+    // Dummy display does nothing in set_pixel
+    r2d2::display::display_dummy_c<r2d2::display::st7735_128x160_s> test_display;
+
+    r2d2::display::module_c module(mock_bus, test_display);
+
+    SECTION("X out of bounds"){
+        constexpr uint8_t start_x = 50;
+        constexpr uint8_t start_y = 75;
+        // Send the cursor to 50,75
+        auto frame_start_pos = mock_bus.create_frame<r2d2::frame_type::CURSOR_POSITION>(
+        {static_cast<uint8_t>(r2d2::claimed_display_cursor::OPEN_CURSOR), start_x, start_y});
+
+        // Actually place the frame on the bus, so the module
+        // can see and process it.
+        mock_bus.accept_frame(frame_start_pos);
+
+        module.process();
+
+        //send cursor to 130,50
+        //130 being out of bounds 50 is within bounds.
+        constexpr uint8_t new_x = 130;
+        constexpr uint8_t new_y = 50;
+
+        auto frame_pos = mock_bus.create_frame<r2d2::frame_type::CURSOR_POSITION>(
+        {static_cast<uint8_t>(r2d2::claimed_display_cursor::OPEN_CURSOR), new_x, new_y});
```

Comment:
Might be me, but the formatting is weird here

Reply:
could add a tab if necesary, did not run clang since it will ruin most of the formatting in this file

Reply:
```suggestion
        auto frame_pos = mock_bus.create_frame<r2d2::frame_type::CURSOR_POSITION>({
            static_cast<uint8_t>(r2d2::claimed_display_cursor::OPEN_CURSOR), new_x, new_y
        });
```